### DISCUSSION TITLE
Waze support

### DISCUSSION THREAD

**Discussion by dkatuwawela** [BimmerGestalt/AAIdrive#682] [Q&A]
First off, phenomenal work! Kudos, and I'm a huge fan!
Is there an ability to integrate Waze similar to how the audio apps work?

**Comment by bogdan-calapod** [comment]
AFAIK this isn't possible, but you can use the Screen Mirroring add-on to use basically any app on your phone

**Comment by hufman** [comment]
This is not possible, indeed. The music apps implement a standard remote control protocol, built into the OS for Android Auto and Android Wear and Bluetooth connections to control them, so AAIdrive is able to use that existing protocol. There isn't a similar protocol to embed other apps for control and screencasting, and I don't see any information on how to embed Waze into another app. Embedding would be required so that AAIdrive could capture the window while the screen is off.
However, that part of the app would be relatively standard Android hacking, not car-specific, so I would love to have help from an Android dev who knows more information!
